### ISSUE TITLE
Interesting examples to reproduce

### ISSUE THREAD

**Issue by tlienart** [tlienart/JSXGraph.jl#6]
Opinionated list of examples that

* [x] https://jsxgraph.org/wiki/index.php/Draggable_exponential_function
  * **doable**
* [ ] https://jsxgraph.org/wiki/index.php/Lagrange_interpolation
  * requires button
* [ ] https://jsxgraph.org/wiki/index.php/Least-squares_circle_fitting
  * requires circle
* [ ] https://jsxgraph.org/wiki/index.php/Least-squares_line_fitting
  * requires line and circle
* [x] https://jsxgraph.org/wiki/index.php/Lituus
  * **doable**
* [ ] https://jsxgraph.org/wiki/index.php/Logarithmic_spiral
  * tangent, glider
* [ ] https://jsxgraph.org/wiki/index.php/Matrix_multiplication , https://jsxgraph.org/wiki/index.php/Matrix_multiplication_II
  * arrows
* [ ] https://jsxgraph.org/wiki/index.php/P-Norm
  * **doable**
* [ ] https://jsxgraph.org/wiki/index.php/Polynomial_regression_I
  * glider, text
* [ ] https://jsxgraph.org/wiki/index.php/Polynomial_regression_II
  * text
* [ ] https://jsxgraph.org/wiki/index.php/Polynomials
  * **doable**
* [ ] https://jsxgraph.org/wiki/index.php/Power_Series_for_sine, https://jsxgraph.org/wiki/index.php/Power_Series_for_sine_and_cosine https://jsxgraph.org/wiki/index.php/Power_series_for_the_exponential_function
  * **doable**
* [ ] https://jsxgraph.org/wiki/index.php/Riemann_integrability , https://jsxgraph.org/wiki/index.php/Riemann_sum_II, https://jsxgraph.org/wiki/index.php/Riemann_sum_III https://jsxgraph.org/wiki/index.php/Riemann_sums
  * riemansum
* [ ] https://jsxgraph.org/wiki/index.php/Inequalities
  * inequality, line
* [ ] https://jsxgraph.org/wiki/index.php/Hilbert_curve
  * **doable** but they use a horrible function to do it... 

(haven't checked what's before letter D)

* [ ] https://jsxgraph.org/wiki/index.php/Differential_equations

### Other stuffs / docs

* [x] https://jsxgraph.org/docs/symbols/Button.html

**Comment by tlienart** [comment]
https://tlienart.github.io/JSXGraphWeb/page2/

**Comment by s-celles** [comment]
> **Note:** This recap was generated by an AI agent (Claude Code) acting on my behalf — please flag any inaccuracies. Each "Done" item below has been independently verified against the 0.5.3 source (commit `338f5b9`), not trusted from the 2020 `[x]` marks.

## Status as of `v0.5.3`

### ✅ Ported to `docs/src/wiki_examples.md`

Seven examples from this checklist now live on the new [**Wiki Examples**](https://juliajsxgraph.github.io/JSXGraph.jl/dev/wiki_examples/) documentation page (deployed once CI finishes). Each one is a self-contained `@example` block that renders an interactive board.

| # | Wiki page | Section in the docs | Notes |
|---|---|---|---|
| 1 | [Draggable exponential function](https://jsxgraph.org/wiki/index.php/Draggable_exponential_function) | [`§1`](https://github.com/JuliaJSXGraph/JSXGraph.jl/blob/338f5b9/docs/src/wiki_examples.md#L11) | Uses the wiki's `A.Y()^(x/A.X())` formula so the anchor point is on the curve by construction. Requires v0.5.3's `@jsf` property-access (`α.X()`, `α.Y()`) support. |
| 2 | [Lituus](https://jsxgraph.org/wiki/index.php/Lituus) | [`§2`](https://github.com/JuliaJSXGraph/JSXGraph.jl/blob/338f5b9/docs/src/wiki_examples.md#L30) | Slider `k`, polar curve `√(k/φ)`. Built as a parametric `curve(fx, fy, t_start, t_end)` because the standalone `polar` returns its own board. |
| 3 | [P-Norm](https://jsxgraph.org/wiki/index.php/P-Norm) | [`§3`](https://github.com/JuliaJSXGraph/JSXGraph.jl/blob/338f5b9/docs/src/wiki_examples.md#L57) | Slider `p`, implicit curve `|x|^p + |y|^p = 1`. Not yet cross-checked against the wiki page; will revisit if it diverges. |
| 4 | [Polynomials](https://jsxgraph.org/wiki/index.php/Polynomials) | [`§4`](https://github.com/JuliaJSXGraph/JSXGraph.jl/blob/338f5b9/docs/src/wiki_examples.md#L75) | Four-slider quartic `a·x⁴ + b·x³ + c·x² + d·x`. Wiki comparison pending. |
| 5 | [Power Series for sine](https://jsxgraph.org/wiki/index.php/Power_Series_for_sine) | [`§5`](https://github.com/JuliaJSXGraph/JSXGraph.jl/blob/338f5b9/docs/src/wiki_examples.md#L94) | Slider `N` for truncation order. Body is a hand-written `JSXGraph.JSFunction` because `@jsf` does not transpile `for`/`while` (`:for` is in `UNSUPPORTED_EXPR_HEADS`); the placeholder marker `__JSF_REF_s__` is substituted at render time via the same mechanism `@jsf` uses internally. |
| 6 | [Riemann sums](https://jsxgraph.org/wiki/index.php/Riemann_sums) | [`§6`](https://github.com/JuliaJSXGraph/JSXGraph.jl/blob/338f5b9/docs/src/wiki_examples.md#L134) | Three sliders (`n`, `start`, `end`). Wraps each slider in `@jsf () -> slider_var` so JSXGraph receives `function(){return el_xxx.Value();}` — required because `functiongraph` and `riemannsum` range slots don't accept raw slider element refs. |
| 7 | [Inequalities](https://jsxgraph.org/wiki/index.php/Inequalities) | [`§7`](https://github.com/JuliaJSXGraph/JSXGraph.jl/blob/338f5b9/docs/src/wiki_examples.md#L184) | Linear half-planes. Adds a new public method [`line(c::Real, b::Real, a::Real)`](https://github.com/JuliaJSXGraph/JSXGraph.jl/blob/338f5b9/src/elements.jl#L185) for the JSXGraph homogeneous-coords form `c + b·x + a·y = 0`. |

### 🍒 Lot 2 — low-hanging fruits (~30 min total)

Direct variants of work already done; no new constructors, no engine changes.

| Wiki page | Pattern to reuse | Tweak |
|---|---|---|
| [Power Series for sine **and cosine**](https://jsxgraph.org/wiki/index.php/Power_Series_for_sine_and_cosine) | §5 — raw `JSFunction` with truncation slider | Add a second `JSFunction` with cos terms `(-1)^k · t^(2k) / (2k)!` |
| [Power series for the exponential](https://jsxgraph.org/wiki/index.php/Power_series_for_the_exponential_function) | §5 | Even simpler: `t^k / k!`, no sign alternation |
| [Riemann sum II / III](https://jsxgraph.org/wiki/index.php/Riemann_sum_II), [Riemann integrability](https://jsxgraph.org/wiki/index.php/Riemann_integrability) | §6 — three-slider sum | Swap `f`, the interval defaults, and the `type` (`"right"`, `"trapezoidal"`, …) |

### 🍐 Lot 3 — moderate (~15–20 min each)

Wider in scope but still reachable with the v0.5.3 API.

| Wiki page | What it needs |
|---|---|
| [Matrix multiplication I](https://jsxgraph.org/wiki/index.php/Matrix_multiplication) | 4 sliders for matrix entries + 1 draggable input vector + `arrow` for the output vector. Output uses `@jsf (x, y) -> a*x + b*y` style functions reading `vec.X()` / `vec.Y()`. |
| [Hilbert curve](https://jsxgraph.org/wiki/index.php/Hilbert_curve) | Recurse in Julia to produce `Vector{Float64}` of `(x, y)` pairs, then feed a data-driven curve through the Tables.jl extension: `plot((x=…, y=…), :x, :y)`. Static (no slider). |
| [Logarithmic spiral](https://jsxgraph.org/wiki/index.php/Logarithmic_spiral) | Pattern from §2 (Lituus parametric form) + a `glider` on the spiral + a `tangent` element through that glider. |

### 🍋 Lot 4+ — defer until a Lot 3 follow-up

Each of these needs either a JS regression call (`JXG.Math.Numerics.*`), a mutable button-triggered state machine, or a full RK4 solver in raw JS. Worth tracking in this issue but not low-effort.

| Wiki page | Blocker |
|---|---|
| [Lagrange interpolation](https://jsxgraph.org/wiki/index.php/Lagrange_interpolation) | `button` controlling a mutable point list; need to manage state in JS. |
| [Polynomial regression I](https://jsxgraph.org/wiki/index.php/Polynomial_regression_I), [II](https://jsxgraph.org/wiki/index.php/Polynomial_regression_II) | Calls `JXG.Math.Numerics.regression…` from inside a raw `JSFunction`. |
| [Least-squares circle](https://jsxgraph.org/wiki/index.php/Least-squares_circle_fitting), [line](https://jsxgraph.org/wiki/index.php/Least-squares_line_fitting) fitting | Same blocker as Polynomial regression. |
| [Matrix multiplication II](https://jsxgraph.org/wiki/index.php/Matrix_multiplication_II) | The extended variant adds eigenvector visualization — wider scope. |
| [Differential equations](https://jsxgraph.org/wiki/index.php/Differential_equations) | Hand-written RK4 in a `JSFunction` body. |

### 🟥 Known limitations exposed during this work

- `@jsf` does not transpile `for` / `while` (see `UNSUPPORTED_EXPR_HEADS` in [`src/jsfunction.jl`](https://github.com/JuliaJSXGraph/JSXGraph.jl/blob/338f5b9/src/jsfunction.jl#L282)). Workaround: build a raw `JSXGraph.JSFunction(code; refs=…)` where the body uses placeholders like `__JSF_REF_s__` for captured element refs (§5 is the reference example).
- `functiongraph` only exposes the single-parent form `functiongraph(f)`. To pass `xmin` / `xmax` as functions or sliders, you currently have to call `JSXGraph._create_element("functiongraph", (f, xmin, xmax), …)` (used in §6). A public `functiongraph(f, xmin, xmax)` constructor would be worth adding.
- JSXGraph's `inequality` element only accepts a `line` parent — for a curve-based inequality (`y ≤ f(x)` for an arbitrary `f`) the upstream API doesn't exist, so the "`y ≤ sin(x)`" framing in earlier drafts of this page was a stretch.

### Suggested next step

Pick up **Lot 2** (the three sine/cos/exp series and a Riemann variant) as a single docs-only follow-up — purely additive, no API risk. **Lot 3** is the right next bite once that lands.
